pkg/reconciler/cloudeventingress/cloudeventingress.go

```diff
@@ -0,0 +1,148 @@
+	"github.com/knative/eventing-sources/pkg/controller/sdk"
+	"github.com/knative/eventing-sources/pkg/controller/sinks"
+	"github.com/knative/eventing-sources/pkg/reconciler/cloudeventingress/resources"
+	"github.com/knative/pkg/logging"
+	servingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	// controllerAgentName is the string used by this controller to identify
+	// itself when creating events.
+	controllerAgentName = "cloudeventingress-controller"
+
+	// raImageEnvVar is the name of the environment variable that contains the receive adapter's
+	// image. It must be defined.
+	raImageEnvVar = "CLOUDINGRESS_RA_IMAGE"
+)
+
+// Add creates a new CloudEventIngress Controller and adds it to the Manager with
+// default RBAC. The Manager will set fields on the Controller and Start it when
+// the Manager is Started.
+func Add(mgr manager.Manager) error {
+	raImage, defined := os.LookupEnv(raImageEnvVar)
+	if !defined {
+		return fmt.Errorf("required environment variable %q not defined", raImageEnvVar)
+	}
+
+	log.Println("Adding the Cron Job Source controller.")
+	p := &sdk.Provider{
+		AgentName: controllerAgentName,
+		Parent:    &v1alpha1.CloudEventIngress{},
+		Owns:      []runtime.Object{&servingv1alpha1.Service{}},
+		Reconciler: &reconciler{
+			scheme:              mgr.GetScheme(),
+			receiveAdapterImage: raImage,
+		},
+	}
+
+	return p.Add(mgr)
+}
+
+// reconciler reconciles a CloudEventIngress object
+type reconciler struct {
+	client              client.Client
+	scheme              *runtime.Scheme
+	receiveAdapterImage string
+}
+
+func (r *reconciler) InjectClient(c client.Client) error {
+	r.client = c
+	return nil
+}
+
+func (r *reconciler) Reconcile(ctx context.Context, object runtime.Object) error {
+	logger := logging.FromContext(ctx)
+
+	src, ok := object.(*v1alpha1.CloudEventIngress)
+	if !ok {
+		logger.Error("could not find CloudEventIngress source", zap.Any("object", object))
+		return nil
+	}
+
+	// This Source attempts to reconcile three things.
+	// 1. Determine the sink's URI.
+	//     - Nothing to delete.
+	// 2. Create a knative service
+
+	src.Status.InitializeConditions()
+
+	sinkURI, err := sinks.GetSinkURI(ctx, r.client, src.Spec.Sink, src.Namespace)
+	if err != nil {
+		src.Status.MarkNoSink("NotFound", "")
+		return err
+	}
+	src.Status.MarkSink(sinkURI)
+
+	err = r.createKnativeService(ctx, src, sinkURI)
+	if err != nil {
+		logger.Error("Unable to create the receive adapter", zap.Error(err))
+		return err
+	}
+	src.Status.MarkDeployed()
+	return nil
+}
+
+func (r *reconciler) createKnativeService(ctx context.Context, src *v1alpha1.CloudEventIngress, sinkURI string) error {
+	// ra, err := r.getReceiveAdapter(ctx, src)
```

Comment:
need to clean up the commented out code.

Reply:
Comment removed.